### PR TITLE
Refs #12634 - PXEGrub 1 and 2 are locked too

### PR DESCRIPTION
**Pull request by lzap**
When https://github.com/theforeman/foreman/pull/3679 is merged, discovery will
no longer work when Lock PXE setting is turned on. This fixes it.

### CODE REVIEW

**Comment by dLobatog** [comment]
[test]

File: app/models/host/discovered.rb [hunk 1]
```diff
@@ -117,9 +117,11 @@ def populate_fields_from_facts(facts = self.facts_hash, ignored_type = nil)
     if Setting::Discovered.discovery_lock? && self.subnet.tftp?
       begin
         @host = self
-        Rails.logger.info "Locking discovered host #{self.mac} in subnet #{subnet}"
-        template = unattended_render(::ProvisioningTemplate.find_by_name(Setting['discovery_lock_template']).template)
-        subnet.tftp_proxy.set mac, :pxeconfig => template
+        TemplateKind::PXE.each do |kind|
+          Rails.logger.info "Locking discovered host #{self.mac} in subnet #{subnet} via #{kind} template"
+          template = unattended_render(::ProvisioningTemplate.find_by_name(Setting["discovery_#{kind}_lock_template"]).template)
```
Comment (dLobatog) [inline app/models/host/discovered.rb]: This is going to fail to find the setting in a test environment, normally we circumvent the issue through fixtures in core. The method has many responsibilities already, if you consider moving this to a separate method I think it'd be nicer.

Comment (dLobatog) [reply app/models/host/discovered.rb]: Stubbing is fine too, probably better as adding fixtures from rails engines is cumbersome

**Comment by dLobatog** [comment]
@lzap notice tests break because of trying to find a Setting. Other than that :+1: , can you create an issue specific for discovery though? Normally projects shouldn't share issue numbers, especially between core and others because the issue numbers are used for releases. 

Thanks!

**Comment by lzap** [comment]
Amended, fixed tests, pulled method out, fixed ref issue number (it was not correct). Should be all fine now.

**Comment by lzap** [comment]
Rebased.

**Comment by lzap** [comment]
Irrelevant test failure, rebased and pushed again.

**Comment by lzap** [comment]
Mind merging @dLobatog ? I refactored tests and I would like to rebase my other PR on top of this one as it touches lots of test files.

**Comment by lzap** [comment]
@dLobatog the reason is http://projects.theforeman.org/issues/16143 (need to rebase it on top of this due to test refactoring in here)

**Comment by dLobatog** [comment]
Thanks @lzap !

**Pull request merged**